### DISCUSSION TITLE
Webview  filter urls?

### DISCUSSION THREAD

**Discussion by l-o-u-i-s-g** [beeware/toga#2940] [Q&A]
Is it possible to filter urls  in the webview?   For example to only allow  urls from  mydomain.com  so that if the user clicks on a link in the page  and the link doesn't match then  display error message or redirect  etc .. 

other than "faking" it with on_webview_loaded

**Comment by freakboy3742** [comment]
There's definitely no public API to enforce a restriction like this.

It's likely possible with platform specific APIs - I believe most of those APIs have a mechanism where you can inject into the "pre-link-visit" logic to allow/reject the link. If someone wanted to provide an implementation exposing a public Toga API for that capability, I wouldn't object - it's a reasonable enough requirement to enforce.

**Comment by t-arn** [comment]
@l-o-u-i-s-g I implemented this for Windows in the TaWebView class https://www.tanapro.ch/products/taTogaLib/docs/html/index.html
